cli/internal/specs/v0/destination.go

```diff
@@ -39,8 +42,8 @@ func (d *Destination) SetDefaults(defaultBatchSize, defaultBatchSizeBytes int) {
 	if d.Spec == nil {
 		d.Spec = make(map[string]any)
 	}
-	if d.Registry.String() == "" {
-		d.Registry = RegistryGithub
+	if d.registryInferred && d.Registry == 0 {
+		d.Registry = RegistryCloudQuery
```

Comment:
The gist part 2, for destinations: if registry was an empty string during unmarshal and defaulted to the zero-value of `int`, we override it to be our default.